rotkehlchen/chain/ethereum/modules/balancer/utils.py

```diff
@@ -0,0 +1,299 @@
+        raise DeserializationError(
+            f'Invalid ethereum address: {token_address} in token price.',
+        ) from e
+
+    return token_address, usd_price
+
+
+def deserialize_token_day_data(
+        raw_token_day_data: Dict[str, Any],
+) -> Tuple[ChecksumEthAddress, Price]:
+    """May raise DeserializationError"""
+    try:
+        token_address = raw_token_day_data['token']['id']
+        usd_price = deserialize_price(raw_token_day_data['priceUSD'])
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    try:
+        token_address = to_checksum_address(token_address)
+    except ValueError as e:
+        raise DeserializationError(
+            f'Invalid ethereum address: {token_address} in token day data.',
+        ) from e
+
+    return token_address, usd_price
+
+
+def calculate_amm_trade_from_amm_swaps(swaps: List[AMMSwap]) -> AMMTrade:
+    """AMMTrade is always a buy.
+
+    [USDC -> AMPL]               BASE_QUOTE pair is AMPL_USDC.
+    [USDC -> AMPL, AMPL -> WETH] BASE_QUOTE pair is WETH_USDC.
+    """
+    if len(swaps) == 0:
+        raise AssertionError("Swaps can't be an empty list")
+
+    amm_trade = AMMTrade(
+        trade_type=TradeType.BUY,
+        base_asset=swaps[-1].token1,
+        quote_asset=swaps[0].token0,
+        amount=swaps[-1].amount1_out,
+        rate=Price(swaps[-1].amount1_out / swaps[0].amount0_in),
+        swaps=swaps,
+        trade_index=0,
+    )
+    return amm_trade
+
+
+def get_trades_from_tx_swaps(swaps: List[AMMSwap]) -> List[AMMTrade]:
+    """Aggregates N AMMSwaps in an AMMTrade.
+
+    We expect that the swaps likely to be aggregated in a single trade are in
+    sequence (the next). This function currently does not have interleaved
+    matching capabilities.
+
+    When swaps are done via the Balancer Exchange Proxy (caller address is
+    `0x3e66b66fd1d0b02fda6c811da9e0547970db2f21`) the previous swap N amount out
+    matches the swap N+1 amount in. However, there may be slightly differences
+    between these amounts when the caller is a custom contract. The former case
+    will be always be a trade with two swaps. On the latter it will depend on
+    the quantities. If they match, a trade with two swaps, otherwise two trades
+    (each one with one swap).
+
+    Aggregation criteria:
+    - AMMSwap N amount1_out == AMMSwap N+1 amount0_in
+    - AMMSwap N token1 == AMMSwap N+1 token0
+    """
+    trades: List[AMMTrade] = []
+    trade_swaps: List[AMMSwap] = []
+    last_idx = len(swaps) - 1
+    for idx, swap in enumerate(swaps):
+        trade_swaps.append(swap)
+        if idx == last_idx:
+            trade = calculate_amm_trade_from_amm_swaps(trade_swaps)
+            trades.append(trade)
+            break
+
+        next_swap = swaps[idx + 1]
+        if (
+            swap.amount1_out != next_swap.amount0_in or
+            swap.token1.ethereum_address != next_swap.token0.ethereum_address
+        ):
```

Comment:
can you put this if into a descriptive variable that will make it easy to understand what you are doing here?

```suggestion
        descriptive_variable = (
            swap.amount1_out != next_swap.amount0_in or
            swap.token1.ethereum_address != next_swap.token0.ethereum_address
        )
        if descriptive_variable:
```